app/controllers/api/base_controller.rb

```diff
@@ -260,6 +247,67 @@ def action_permission
       end
     end
 
+    def parent_permission(child_permission)
+      case child_permission.to_s
+      when 'create', 'destroy'
+        'edit'
+      when 'edit', 'view'
+        'view'
+      else
+        raise ::Foreman::Exception.new(N_("unknown parent permission for %s"), "#{params[:controller]}##{child_permission}")
+      end
+    end
+
+    def parent_resource_details
+      classes_arr = params.map do |key, value|
```

Comment:
please split this method into two, one that finds the parent resource parameter, second that will construct scope (which involves resource_class_for and resource_name_for)